src/command_modules/azure-cli-consumption/azure/cli/command_modules/consumption/commands.py

```diff
@@ -7,24 +7,44 @@
 from azure.cli.command_modules.consumption._transformers import transform_usage_list_output
 from azure.cli.command_modules.consumption._transformers import transform_reservation_summaries_list_output
 from azure.cli.command_modules.consumption._transformers import transform_reservation_details_list_output
+from azure.cli.command_modules.consumption._transformers import transform_pricesheet_get_output
 from azure.cli.command_modules.consumption._client_factory import usage_details_mgmt_client_factory
 from azure.cli.command_modules.consumption._client_factory import reservations_summaries_mgmt_client_factory
 from azure.cli.command_modules.consumption._client_factory import reservations_details_mgmt_client_factory
+from azure.cli.command_modules.consumption._client_factory import pricesheet_mgmt_client_factory
 from ._exception_handler import consumption_exception_handler
 from ._validators import validate_both_start_end_dates
 from ._validators import validate_reservations_summaries
 from ._validators import validate_reservations_details
+from ._validators import validate_pricesheet
+from ._validators import validate_usage_bp_inputs
 
 
 def load_command_table(self, _):
     with self.command_group('consumption usage') as g:
         g.custom_command('list', 'cli_consumption_list_usage', transform=transform_usage_list_output,
                          exception_handler=consumption_exception_handler, validator=validate_both_start_end_dates, client_factory=usage_details_mgmt_client_factory)
 
+        g.custom_command('billing period list', 'cli_consumption_list_usage_by_billing_period', transform=transform_usage_list_output,
+                         exception_handler=consumption_exception_handler, validator=validate_usage_bp_inputs, client_factory=usage_details_mgmt_client_factory)
+
     with self.command_group('consumption reservations summaries') as s:
         s.custom_command('list', 'cli_consumption_list_reservations_summaries', transform=transform_reservation_summaries_list_output,
                          exception_handler=consumption_exception_handler, validator=validate_reservations_summaries, client_factory=reservations_summaries_mgmt_client_factory)
 
+        s.custom_command('reservation id list', 'cli_consumption_list_reservations_summaries_reservation_id', transform=transform_reservation_summaries_list_output,
```

Comment:
Same here.

Reply:
'reservation id list' is a single subgroup under consumption reservation summaries - given the best practices doc, looks like 'reservation-id-list' is a better choice?